kernel/kernel-api/src/main/java/io/delta/kernel/internal/TransactionBuilderImpl.java

```diff
@@ -213,6 +215,11 @@ protected TransactionImpl buildTransactionInternal(
         snapshot -> validateWriteToExistingTable(engine, snapshot, isCreateOrReplace));
     validateTransactionInputs(engine, isCreateOrReplace);
 
+    final Committer committer =
+        latestSnapshot
+            .map(SnapshotImpl::getCommitter)
```

Comment:
Will it be problematic if a catalog owned cc table falling back to FileSystemManagedTableOnlyCommitter? (i.e. due to bug)

Reply:
Yes -- engines must inject the right committer